### PR TITLE
Implement LeafletPortrayal dataclass for GeoAgent portrayal

### PR DESCRIPTION
**Pull request by wang-boyu**
Fixes https://github.com/projectmesa/mesa-geo/issues/79.

### CODE REVIEW

File: mesa_geo/visualization/modules/MapVisualization.py [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
 
+@dataclass
+class LeafletPortrayal:
+    """A dataclass defining the portrayal of a GeoAgent in Leaflet map.
+
+    The fields are defined to be consistent with GeoJSON options in
+    Leaflet.js: https://leafletjs.com/reference.html#geojson
+    """
+
+    style: dict = None
```
Comment (rht) [inline mesa_geo/visualization/modules/MapVisualization.py:20]: Should this be `Dict[str, Any] | None`? Or do the values have a known type?

Comment (wang-boyu) [reply mesa_geo/visualization/modules/MapVisualization.py:20]: It seems to be `Dict[str, str | bool | int | float] | None`, but `Dict[str, Any] | None` is neater and more flexible. I'll change to this.

Comment (wang-boyu) [reply mesa_geo/visualization/modules/MapVisualization.py]: Changed to `Dict[str, Any] | None`.

File: mesa_geo/visualization/modules/MapVisualization.py [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
 
+@dataclass
+class LeafletPortrayal:
+    """A dataclass defining the portrayal of a GeoAgent in Leaflet map.
+
+    The fields are defined to be consistent with GeoJSON options in
+    Leaflet.js: https://leafletjs.com/reference.html#geojson
+    """
+
+    style: Dict[str, Any] | None = None
```
Comment (rht) [inline mesa_geo/visualization/modules/MapVisualization.py:23]: I think defining `str | bool | int | float` as a Mypy type is better than `Any`.

Comment (wang-boyu) [reply mesa_geo/visualization/modules/MapVisualization.py:23]: Do you mean a type alias such as `LeafletOption = Union[str, bool, int, float]`?

Comment (rht) [reply mesa_geo/visualization/modules/MapVisualization.py:23]: Yes, unfortunately it has to use `Union` for now.

**Pull request merged**